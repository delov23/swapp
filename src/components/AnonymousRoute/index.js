export { default as AnonymousRoute } from './AnonymousRoute';
