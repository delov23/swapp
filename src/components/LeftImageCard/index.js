export { default as LeftImageCard } from './LeftImageCard';
