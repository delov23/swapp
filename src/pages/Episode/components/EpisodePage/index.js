export { default as EpisodePage } from './EpisodePage';
