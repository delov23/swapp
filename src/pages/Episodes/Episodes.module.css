.cards {
  display: grid;
  grid-template-columns: repeat(auto-fill, 20rem);
  justify-content: center;
  margin: auto;
}

@media screen and (max-width: 600px) {
  .cards {
    grid-template-columns: repeat(auto-fill, 22rem);
  }
}
