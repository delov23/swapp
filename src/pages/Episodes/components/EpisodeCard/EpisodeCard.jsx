import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from './EpisodeCard.module.css';
import { themify } from '../../../../utils/themify';
import ThemeContext from '../../../../context/ThemeContext';

const EpisodeCard = ({ episode: { image, id, title, openingCrawl } }) => {
  const theme = useContext(ThemeContext);

  return (
    <Link to={'/episodes/' + id} className={styles[themify('card', theme)]}>
      <img className={styles.cardImg} src={image} alt="" />
      <h3 className={styles[themify('heading', theme)]}>{title}</h3>
      <p className={styles.paragraph}>{openingCrawl}</p>
    </Link>
  );
};

EpisodeCard.propTypes = {
  episode: PropTypes.object.isRequired,
};

export default EpisodeCard;
