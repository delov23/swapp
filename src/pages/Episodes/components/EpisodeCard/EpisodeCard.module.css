.card {
  display: flex;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  width: 18rem;
  border-radius: 10px;
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  -ms-border-radius: 10px;
  -o-border-radius: 10px;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: 2rem;
  flex-basis: 30%;
}

.cardLight {
  composes: card;
  border-color: #e5e9f2;
  background-color: white;
}

.cardDark {
  composes: card;
  border-color: #3c4858;
  background-color: #333;
}

.cardImg {
  border: none;
  width: 100%;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

.heading {
  font-family: Starwars;
  font-weight: 100;
  margin: 0;
  padding-top: 2rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.headingLight {
  composes: heading;
  color: var(--mainCyan);
}

.headingDark {
  composes: heading;
  color: var(--mainYellow);
}

.paragraph {
  padding-left: 1rem;
  padding-right: 1rem;
}

@media screen and (max-width: 800px) {
  .card {
    width: 19.5rem;
  }
}