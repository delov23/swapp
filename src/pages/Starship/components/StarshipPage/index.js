export { default as StarshipPage } from './StarshipPage';
