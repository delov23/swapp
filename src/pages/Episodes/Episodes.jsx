import React from 'react';
import { gql } from 'graphql.macro';
import { useQuery } from '@apollo/react-hooks';

import styles from './Episodes.module.css';
import EpisodeCard from './components/EpisodeCard/EpisodeCard';
import { Loading } from '../../components/Loading/';

export const ALL_EPISODES_QUERY = gql`
  query {
    allEpisodes(first: 100) {
      edges {
        node {
          id
          image
          title
          openingCrawl
        }
      }
    }
  }
`;

const Episodes = () => {
  const { data, loading, error } = useQuery(ALL_EPISODES_QUERY);

  if (loading) return <Loading />;
  if (error) return <h3>Error</h3>;

  return (
    <main className={styles.deck}>
      <div className={styles.cards}>
        {data.allEpisodes.edges.map(({ node }) => (
          <EpisodeCard key={node.id} episode={node} />
        ))}
      </div>
    </main>
  );
};

export default Episodes;
