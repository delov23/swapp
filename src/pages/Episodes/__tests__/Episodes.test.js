import React from 'react';
import { mount, shallow } from 'enzyme';
import { MockedProvider } from '@apollo/react-testing';
import { BrowserRouter as Router, Redirect } from 'react-router-dom';
import wait from 'waait';

import { Episodes } from '../';
import { ALL_EPISODES_QUERY } from '../Episodes';
import { EpisodeCard } from '../components/EpisodeCard';
import { act } from 'react-dom/test-utils';

const mocks = [
  {
    request: {
      query: ALL_EPISODES_QUERY,
    },
    result: () => ({
      data: {
        allEpisodes: {
          edges: [
            {
              node: {
                id: 'episodes.1',
                title: 'One',
                openingCrawl: 'Onee',
                image: 'one',
              },
            },
            {
              node: {
                id: 'episodes.2',
                title: 'Two',
                openingCrawl: 'Twee',
                image: 'two',
              },
            },
          ],
        },
      },
    }),
  },
];

const errorMocks = [
  {
    request: {
      query: ALL_EPISODES_QUERY,
    },
    result: new Error('Test'),
  },
];

describe('<Episodes />', () => {
  it('should match snapshot', () => {
    expect(
      shallow(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Episodes />
        </MockedProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('should display EpisodeCards', async () => {
    let wrapper;

    await act(async () => {
      wrapper = mount(
        <Router>
          <MockedProvider mocks={mocks} addTypename={false}>
            <Episodes />
          </MockedProvider>
        </Router>,
      );

      await wait(0);
      wrapper.update();
    });

    expect(wrapper.find(EpisodeCard)).toHaveLength(2);
  });

  it('should redirect when error occurs', async () => {
    let wrapper;

    await act(async () => {
      wrapper = mount(
        <Router>
          <MockedProvider mocks={errorMocks} addTypename={false}>
            <Episodes />
          </MockedProvider>
        </Router>,
      );

      await wait(0);
      wrapper.update();
    });

    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
});
