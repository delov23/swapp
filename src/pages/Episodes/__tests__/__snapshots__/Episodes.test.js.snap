// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episodes /> should match snapshot 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": false,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": false,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": MockLink {
        "addTypename": false,
        "mockedResponsesByKey": Object {
          "{\\"query\\":\\"{\\\\n  allEpisodes(first: 100) {\\\\n    edges {\\\\n      node {\\\\n        id\\\\n        image\\\\n        title\\\\n        openingCrawl\\\\n      }\\\\n    }\\\\n  }\\\\n}\\\\n\\"}": Array [
            Object {
              "request": Object {
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "first",
                                },
                                "value": Object {
                                  "kind": "IntValue",
                                  "value": "100",
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "allEpisodes",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "edges",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "node",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                            },
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "image",
                                              },
                                            },
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "title",
                                              },
                                            },
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "openingCrawl",
                                              },
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 166,
                    "source": Object {
                      "body": "
  query {
    allEpisodes(first: 100) {
      edges {
        node {
          id
          image
          title
          openingCrawl
        }
      }
    }
  }
",
                      "locationOffset": Object {
                        "column": 1,
                        "line": 1,
                      },
                      "name": "GraphQL request",
                    },
                    "start": 0,
                  },
                },
              },
              "result": [Function],
            },
          ],
        },
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {
            "{\\"query\\":\\"{\\\\n  allEpisodes(first: 100) {\\\\n    edges {\\\\n      node {\\\\n        id\\\\n        image\\\\n        title\\\\n        openingCrawl\\\\n      }\\\\n    }\\\\n  }\\\\n}\\\\n\\"}": Array [
              Object {
                "request": Object {
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "first",
                                  },
                                  "value": Object {
                                    "kind": "IntValue",
                                    "value": "100",
                                  },
                                },
                              ],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "allEpisodes",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "edges",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "node",
                                          },
                                          "selectionSet": Object {
                                            "kind": "SelectionSet",
                                            "selections": Array [
                                              Object {
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "id",
                                                },
                                              },
                                              Object {
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "image",
                                                },
                                              },
                                              Object {
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "title",
                                                },
                                              },
                                              Object {
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "openingCrawl",
                                                },
                                              },
                                            ],
                                          },
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 166,
                      "source": Object {
                        "body": "
  query {
    allEpisodes(first: 100) {
      edges {
        node {
          id
          image
          title
          openingCrawl
        }
      }
    }
  }
",
                        "locationOffset": Object {
                          "column": 1,
                          "line": 1,
                        },
                        "name": "GraphQL request",
                      },
                      "start": 0,
                    },
                  },
                },
                "result": [Function],
              },
            ],
          },
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.3",
      "watchQuery": [Function],
    }
  }
>
  <Episodes />
</ApolloProvider>
`;
